packages/lesswrong/components/hooks/useCookiesWithConsent.ts

```diff
@@ -42,24 +46,36 @@ export function useCookiePreferences(): {
 
   // If the user had not given explicit consent, but the value of COOKIE_PREFERENCES_COOKIE is different to what we are
   // using in the code (fallbackPreferences), update the cookie. This is so that Google Tag Manager handles it correctly.
-  // TODO this is causing an infinite loop somehow, reenable once we figure out why.
-  // useEffect(() => {
-  //   if (explicitConsentRequired === "unknown" || explicitConsentGiven) return;
+  const autoUpdateCount = useRef<number>(0);
+  useEffect(() => {
+    if (explicitConsentRequired === "unknown" || explicitConsentGiven) return;
+
+    if (isValidCookieTypeArray(preferencesCookieValue) && JSON.stringify(cookiePreferences) !== JSON.stringify(preferencesCookieValue)) {
```

Comment:
I'm hoping the check for `isValidCookieTypeArray ` here will be the thing which actually stops the infinite loop from happening